state-chain/chains/src/sol/api.rs

```diff
@@ -148,13 +158,28 @@ impl<Environment: SolanaEnvironment> SolanaApi<Environment> {
 		let vault_program_data_account =
 			Environment::lookup_account(SolanaEnvAccountLookupKey::VaultProgramDataAccount)?;
 		let system_program_id = SolAddress::from_str(SYSTEM_PROGRAM_ID)
-			.expect("Preset System Program ID account must be valid.");
+			.expect("Preset Solana System Program ID account must be valid.");
 		let (nonce_account, durable_nonce) = Environment::nonce_account()?;
 		let compute_price = Environment::compute_price()?;
+		let token_mint_pubkey =
+			Environment::lookup_account(SolanaEnvAccountLookupKey::TokenMintPubkey)?;
+		let token_program_id = SolAddress::from_str(TOKEN_PROGRAM_ID)
+			.expect("Preset Solana Token program ID must be valid.");
+		let token_vault_ata = Environment::lookup_account(
+			SolanaEnvAccountLookupKey::TokenVaultAssociatedTokenAccount,
+		)?;
```

Comment:
I think Roy mentioned this in another PR: it seems like it would make sense for the Solana environment lookup to just expose a single `struct SolanaAccounts { token_program, vault_account, etc etc }` rather than looking up each account individually. 

Doing it individually is (a) hard to read and (b) probably inefficient since we incur the cost of ~1 read per account instead of a single read for all accounts.

Reply:
yes definitely. When we build the environment we can do that.